services/datamanager/builtin/builtin_sync_test.go

```diff
@@ -592,6 +593,20 @@ func TestArbitraryFileUpload(t *testing.T) {
 			serviceFail:          true,
 			connStateConstructor: ConnToConnectivityStateReady,
 		},
+		{
+			name:                 "uploading an image to a dataset with scheduled sync of arbitrary files should work",
```

Comment:
Can we add a test that calls ONLY `UploadBinaryData`? The only reason I’m calling this out is that we _could_ change `UploadImage` so that it works differently and I don’t wanna not have a test that checks this.

Reply:
Ah, no! This is fine. I just wanted to distinguish between the two calls. Sorry about that.